src/hooks/useRequestParams.ts

```diff
@@ -0,0 +1,68 @@
+import { useMemo } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+
+const LEGACY_SEARCH_PARAMS_KEYS = ["chainId", "requester", "timestamp", "identifier", "ancillaryData"] as const;
+const NEW_SEARCH_PARAMS_KEYS = ["chainId", "transactionHash", "eventIndex"] as const;
+
+type LegacyRequestParams = {
+	requester: string;
+	identifier: string;
+	ancillaryData: string;
+	chainId: number;
+	timestamp: number;
+}
+type NewRequestParams = {
+	chainId: number;
+	transactionHash: string;
+	eventIndex: number | undefined;
+}
+
+type RequestParams<IsLegacy extends boolean> = IsLegacy extends true ? LegacyRequestParams : NewRequestParams;
+type Keys<IsLegacy extends boolean> = IsLegacy extends true ? typeof LEGACY_SEARCH_PARAMS_KEYS : typeof NEW_SEARCH_PARAMS_KEYS;
+type UseRequestParamsReturn<IsLegacy extends boolean> = { request: RequestParams<IsLegacy>, error: undefined } | { request: undefined, error: InvalidURLParamsError };
+export function useRequestParams(): UseRequestParamsReturn<false>;
+export function useRequestParams<T extends boolean>(isLegacy: T): UseRequestParamsReturn<T>;
+export default function useRequestParams<T extends boolean>(isLegacy?: T): UseRequestParamsReturn<T> {
```

Comment:
again this would make so much more sense to me as a pure function, since the only thing you need as an input is `searchParams`